cpp/src/traversal/bfs_kernels.cuh

```diff
@@ -844,7 +851,7 @@ __global__ void topdown_expand_kernel(
         for (int iv = 0; iv < TOP_DOWN_BATCH_SIZE; ++iv) {
           IndexType u = vec_u[iv];
           // row_ptr for this vertex origin u
-          vec_row_ptr_u[iv] = (u != -1) ? row_ptr[u] : -1;
+          vec_row_ptr_u[iv] = (u != ~IndexType(0)) ? row_ptr[u] : ~IndexType(0);
```

Comment:
invalid_vertex_id<IndexType>::value is much more readable than ~IndexType(0)

Reply:
Done.